dulwich/objects.py

```diff
@@ -1072,7 +1073,7 @@ def lookup_path(self, lookup_obj, path):
         :param path: Path to lookup
         :return: A tuple of (mode, SHA) of the resulting path.
         """
-        parts = path.split(b'/')
+        parts = path.split(str.encode(os.path.sep))
```

Comment:
Paths here are always git-style paths with a / - it's the responsibility of the caller to convert them.